spec/src/main/asciidoc/servlet-spec-body.adoc

```diff
@@ -587,13 +587,13 @@ becomes available, the thread handling that event continues processing
 either in the same thread or by dispatching to a resource in the
 container using the `AsyncContext`.
 
-Jakarta EE features such as
-<<_Web_Application_Environment>> and
-<<Propagation of Security Identity in EJB™ Calls>> are available only to
-threads executing the initial request or when the request is dispatched
-to the container via the `AsyncContext.dispatch` method. Jakarta EE features
-may be available to other threads operating directly on
-the response object via the `AsyncContext.start(Runnable)` method.
+Jakarta EE features such as <<_Web_Application_Environment>> and
+<<Propagation of Security Identity in Jakarta Enterprise Bean Calls>>
+are available only to threads executing the initial request or when
```

Comment:
Does "initial request" here include `RequestDispatcher.forward` and `RequestDispatcher.include`?   I would think so, but we should probably say explicitly or at least with broader language that doesn't feel like "initial dispatch".   Also any error dispatches are not the initial request, but I also think EE features are available.   

But more importantly for asynchronous processing, there are also async IO callbacks: onDataAvailable, onWritePossible, onAllDataRead and onError that should also have EE features enabled in them.

Rather than enumerating APIs, would it not be better just to say that all container dispatched threads associated with a request should have the EE features enabled?

Reply:
Greg, if you want, can you open a new issue for that?